content/docs/engine-dev/assets/filetypes/texture-support.md

```diff
@@ -0,0 +1,113 @@
+
+* Update materials to reference images via source asset UUID, or product asset asset ID.
+* Build a migration tool that users can run that will update all existing materials on their project to the new material format, so they don't have to manually re-open and re-save all materials.
+* In the guidance for enabling meta sidecar files for images, explain this situation to users, and ask them to run this tool before enabling meta files for image types.
+
+This will mean that material references to images will be stable across image renaming, because materials will use stable IDs.
+### Scene file source assets do not emit job dependencies on images reference by path
+
+In the previous section, authored material source assets had relative path references to image source assets, and they emitted job dependencies on those images. This meant that if the image was renamed with ID stabilization, the job dependency would be the same and the job would re-run, causing the product asset reference to instantly become an error. This is good because it doesn't hide an error for later.
+
+FBX and other scene files also reference source image assets via path, and there's nothing we can do about that. It's a known limitation.
+
+However, the gotcha here is on cache consistency: Right now, without a job dependency, when renaming an image like in the previous section, the FBX file referencing the image will not reprocess. The product asset azmaterial of the FBX file references the product asset via asset ID, but the source asset FBX referenced via path. So now with the rename, the stale azmaterial product asset of the FBX file has a valid asset ID referencing the relocated asset. This means that an asset relocation of an image file may appear to work immediately: rename `CarBody.png` to `CarBody1.png`, and the `Car.azmaterial` product asset of `Car.fbx` appears to render correctly with the asset ID based reference to `CarBody1.streamingimage`. Then, later on someone makes a change to cause `Car.fbx` to reprocess and now the image can't be found and there is an error or warning processing `Car.fbx`. If a user then panics and tries to roll back this change to `Car.fbx`, it will still appear broken, because they had previously been in an unstable state.
+#### Why is this concern?
+
+Cache consistency is very important. Situations like this can break user trust in asset processing, create situations where projects work differently on different team members machines, and the time between causing the issue (renaming the texture) and discovering the bug (the reference is broken, the scene builder can't find the texture and the azmaterial product asset no longer has a valid texture) may take a long time. If everyone on the team is working with iterative caches day to day, and someone finally clears their cache or sets up a new project several weeks after the rename, that's a lot of other changes that could have caused the problem.
+#### What can we do to address this?

```

Comment:
```suggestion
Cache consistency is very important. Situations like this can break user trust in asset processing, create situations where projects work differently on different team members machines, and the time between causing the issue (renaming the texture) and discovering the bug (the reference is broken, the scene builder can't find the texture and the azmaterial product asset no longer has a valid texture) may take a long time. If everyone on the team is working with iterative caches day to day, and someone finally clears their cache or sets up a new project several weeks after the rename, that's a lot of other changes that could have caused the problem.

#### What can we do to address this?

```